var/spack/repos/builtin/packages/libcatalyst/package.py

```diff
@@ -12,9 +12,12 @@ class Libcatalyst(CMakePackage):
 
     homepage = "https://gitlab.kitware.com/paraview/catalyst"
     git = "https://gitlab.kitware.com/paraview/catalyst.git"
+    url = "https://gitlab.kitware.com/api/v4/projects/paraview%2Fcatalyst/packages/generic/catalyst/v2.0.0/catalyst-v2.0.0.tar.gz"
```

Comment:
This rule shows "not-found".

Reply:
2.0.0 is in rc3, so there isn't a final. Does the replacement logic know how to deal with `-rc3` in template URLs?

Reply:
Right, `s/2.0.0/2.0.0-rc3`pointed me to an existing url

Reply:
Yes the replacement logic would know how to handle this

Reply:
@vicentebolea Would that have made a difference for your check?

Reply:
@tldahlgren definitely, checking the url of a PR containing a url change with spack checksum would have validated this PR. I will use it next time for this situations, thanks for sharing your workflow :)

Reply:
Alternatively you can temporarily add the new/changed URL but I have a script to simplify the fetch/checkout process for me.

Reply:
I have used the gh cli utility in this case, its a great help in situations like this